test/extensions/filters/http/aws_lambda/aws_lambda_transformer_test.cc

```diff
@@ -0,0 +1,300 @@
+  EXPECT_EQ(Http::FilterDataStatus::Continue, filter_->decodeData(data2, true));
+
+  EXPECT_EQ(
+    transformedxAmzDateHeader[0]->value().getStringView(),
+    headers.get(Http::LowerCaseString("x-amz-date"))[0]->value().getStringView()
+  );
+  EXPECT_EQ(
+    transformedAuthorizationHeader[0]->value().getStringView(),
+    headers.get(Http::LowerCaseString("authorization"))[0]->value().getStringView()
+  );
+}
+
+TEST_F(AWSLambdaTransformerTest, TestConfigureRequestTransformerSignatureNoBody) {
+  // setup to use the request transformer
+  setupRoute(false, true);
+  Http::TestRequestHeaderMapImpl headers{{":method", "GET"},
+                                         {":authority", "www.solo.io"},
+                                         {":path", "/getsomething"}};
+  
+  EXPECT_EQ(Http::FilterHeadersStatus::StopIteration,
+            filter_->decodeHeaders(headers, false));
+
+  Buffer::OwnedImpl data("");
+
+
+  time_system_.setSystemTime(std::chrono::milliseconds(1000000000000));
+  EXPECT_EQ(Http::FilterDataStatus::Continue, filter_->decodeData(data, true));
+
+  auto transformedAuthorizationHeader = headers.get(Http::LowerCaseString("authorization"));
+  auto transformedxAmzDateHeader = headers.get(Http::LowerCaseString("x-amz-date"));
+
+
+  EXPECT_EQ(transformedxAmzDateHeader[0]->value().getStringView(), "20010909T014640Z");
+  EXPECT_EQ(transformedAuthorizationHeader[0]->value().getStringView(), "AWS4-HMAC-SHA256 Credential=access key/20010909/us-east-1/lambda/aws4_request, SignedHeaders=content-type;host;x-amz-date;x-amz-invocation-type;x-amz-log-type, Signature=6ac2a229351a686ff5a93deb46896f2128a6800a885ef5324c0c0685aa21d786");
+
+  // now, setup to use no transformer
+  setupRoute(false, false);
+  EXPECT_EQ(Http::FilterHeadersStatus::StopIteration,
+            filter_->decodeHeaders(headers, false));
+
+  // the body will be the same as the post-transformer body
+  Buffer::OwnedImpl data2("test body from fake transformer");
+
+  time_system_.setSystemTime(std::chrono::milliseconds(1000000000000));
+  EXPECT_EQ(Http::FilterDataStatus::Continue, filter_->decodeData(data2, true));
+
+  EXPECT_EQ(
+    transformedxAmzDateHeader[0]->value().getStringView(),
+    headers.get(Http::LowerCaseString("x-amz-date"))[0]->value().getStringView()
+  );
+
+  EXPECT_EQ(
+    transformedAuthorizationHeader[0]->value().getStringView(),
+    headers.get(Http::LowerCaseString("authorization"))[0]->value().getStringView()
+  );
+}
+
+TEST_F(AWSLambdaTransformerTest, TestConfigureResponseTransformer){
+  setupRoute(true, false);
+  auto request_headers = setup_encode();
+
+  Buffer::OwnedImpl buf{};
+  auto on_buf_mod = [&buf](std::function<void(Buffer::Instance&)> cb){cb(buf);};
+  EXPECT_CALL(filter_encode_callbacks_, encodingBuffer).WillOnce(Return(&buf));
+  EXPECT_CALL(filter_encode_callbacks_, modifyEncodingBuffer)
+      .WillOnce(Invoke(on_buf_mod));
+
+  auto edResult = filter_->encodeData(buf, true);
```

Comment:
I'm surpised you are not hitting this assert
https://github.com/envoyproxy/envoy/blob/7cddecb88ed2588735e1ae3d94a96893b9986b58/source/common/buffer/buffer_impl.cc#L313

to be extra cautious i suggest:
```suggestion
  Buffer::OwnedImpl dataBuf;
  auto edResult = filter_->encodeData(dataBuf, true);
```